packages/jellyfish-transaction-builder/__tests__/txn/txn_builder_governance_createcfp.test.ts

```diff
@@ -0,0 +1,120 @@
+import { getProviders, MockProviders } from '../provider.mock'
+import { P2WPKHTransactionBuilder } from '../../src'
+import { calculateTxid, fundEllipticPair, sendTransaction } from '../test.utils'
+import { OP_CODES } from '@defichain/jellyfish-transaction'
+import { WIF } from '@defichain/jellyfish-crypto'
+import BigNumber from 'bignumber.js'
+import { GovernanceMasterNodeRegTestContainer } from '../../../jellyfish-api-core/__tests__/category/governance/governance_container'
+import { governance } from '@defichain/jellyfish-api-core'
+import { TxnBuilderError } from '../../src/txn/txn_builder_error'
+
+describe('createCfp', () => {
+  let providers: MockProviders
+  let builder: P2WPKHTransactionBuilder
+  const testing = Testing.create(new GovernanceMasterNodeRegTestContainer())
+
+  beforeAll(async () => {
+    await testing.container.start()
+    await testing.container.waitForWalletCoinbaseMaturity()
+
+    providers = await getProviders(testing.container)
+    providers.setEllipticPair(WIF.asEllipticPair(GenesisKeys[0].owner.privKey)) // set it to container default
+    builder = new P2WPKHTransactionBuilder(providers.fee, providers.prevout, providers.elliptic)
+
+    await providers.randomizeEllipticPair()
+    await testing.container.waitForWalletBalanceGTE(11)
+    await fundEllipticPair(testing.container, providers.ellipticPair, 50)
+    await providers.setupMocks()
+  })
+
+  afterAll(async () => {
+    await testing.container.stop()
+  })
+
+  it('should createCfp', async () => {
+    const script = await providers.elliptic.script()
+    const createCfp = {
+      type: 0x01,
+      title: 'Testing new community fund proposal',
+      amount: new BigNumber(100),
+      address: {
+        stack: [
+          OP_CODES.OP_HASH160,
+          OP_CODES.OP_PUSHDATA_HEX_LE('8b5401d88a3d4e54fc701663dd99a5ab792af0a4'),
+          OP_CODES.OP_EQUAL
+        ]
+      },
+      cycles: 2
+    }
+    const txn = await builder.governance.createCfp(createCfp, script)
+
+    const encoded: string = OP_CODES.OP_DEFI_TX_CREATE_CFP(createCfp).asBuffer().toString('hex')
+    const expectedRedeemScript = `6a${encoded}`
+
+    const outs = await sendTransaction(testing.container, txn)
+    expect(outs[0].value).toEqual(1)
+    expect(outs[0].scriptPubKey.hex).toEqual(expectedRedeemScript)
+
+    const listProposals = await testing.rpc.governance.listProposals()
+    const txid = calculateTxid(txn)
+
+    const proposal = listProposals.find(el => el.proposalId === txid)
+    expect(proposal).toStrictEqual({
+      proposalId: txid,
+      title: createCfp.title,
+      type: governance.ProposalType.COMMUNITY_FUND_REQUEST,
+      status: governance.ProposalStatus.VOTING,
+      amount: createCfp.amount,
+      cyclesPaid: 1,
+      totalCycles: createCfp.cycles,
+      finalizeAfter: expect.any(Number),
+      payoutAddress: '2N5wvYsWcAWQUed5vfPxopxZtjkqoT8dFM3'
+    })
+  })
+
+  it('should reject with invalid type', async () => {
```

Comment:
Do you want to validate cycle?

'should not createCfp with cycles < 1'
'should not createCfp with cycle > 3'

Like what you did in your rpc?